src/workflows-model.ts

```diff
@@ -192,6 +192,36 @@ export interface JobDefaults {
   readonly run?: RunSettings;
 }
 
+/**
+ * The type of runner to run the job on. Can be GitHub or Self-hosted.
+ * In case of self-hosted, a list of labels can be supplied.
+ */
+export class Runner {
+  public static readonly UBUNTU_LATEST = new Runner(['ubuntu-latest']);
+
+  public static readonly WINDOWS_LATEST = new Runner(['windows-latest']);
+
+  public static readonly MACOS_LATEST = new Runner(['macos-latest']);
+
+  public static selfHosted(labels: string[]): Runner {
```

Comment:
All these public APIs need docs. Specifically for `selfHosted`, please mention that you can supply `self-hosted` as a label, but you do not have to.

Reply:
Added comment to code and mentioned in README